addon-test-support/qunit-configuration.js

```diff
@@ -0,0 +1,3 @@
+import * as QUnit from 'qunit';
```

Comment:
what happens here if the app is using Mocha instead of QUnit?

Reply:
I was hoping it would be undefined, just tried it, it fails to link.

So I either have to find a way to check if qunit it installed and then conditionally import.

While it would be nice to do it automatically, Qunit5 is already making people edit the index.html so we could document adding the following lines to test-helper.js for qunit5

```javascript
import QUnit from 'qunit';

QUnit.config.urlConfig.push({ id: 'mirageLogging', label: 'Mirage logging', });
```

Does mocha in some way provide for adding this option to its UI? Mirage has code to add it for qunit, but I dont think there is code for mocha. Should the add-qunit-option.js code be removed from mirage to make it test framework independant?

Reply:
> Does mocha in some way provide for adding this option to its UI? 

nope

> Should the add-qunit-option.js code be removed from mirage to make it test framework independant?

I don't think we should remove it. It brings quite a bit of value to QUnit users, and just because Mocha does not support it shouldn't mean that QUnit users also can't enjoy it.

> So I have to find a way to check if qunit it installed and then conditionally import.

yeah, I think that would be the way to go.

I think something like https://github.com/simplabs/qunit-dom/blob/a0aa749cf9d561ea5cf331fb8c50271d666b68e7/tests/test-helper.js#L7-L13 might be able to solve this issue

Reply:
The problem is the import statement not the setup code. In your example the import for quint is outside the IF. If qunit is not defined the import fails and will never get to the if.  

That being the case it would be easier it ember-qunit used the dependencySatisfies on mirage and did the add, since we know qunit is there.

The only solution I see is the user opting into this by documenting adding a couple lines. OR.... we could  put the new lines in the mirage/config.js as part of the template, and if you are using mocha document that you have to remove them.

What we need is a conditional import, Im just not good enough at modules to know how to do that. Can google some.

Reply:
sorry, I wasn't explaining properly. I was wondering if we could use something roughly like:

```js
if (dependencySatisfies('ember-qunit', '*')) {
  const QUnit = importSync('ember-qunit');
  doSomething(QUnit);
}
```

Reply:
Originally the code was in ember-cli-build.js, when I looked there QUNIT was no longer a global. Thats where the need to import came from. To fix the problem I moved the code from ember-cli-build.js to test-support/index.js.   After more testing it turns out it is a global again there (I would assume making qunit a peer-dependancy caused a load order problem and it was no longer a global in ember-cli-build.js but was when test-support/index.js got around to executing). So I could have just gone with the original code 
```javascript
if (typeof QUnit !== 'undefined') {
    QUnit.config.urlConfig.push({
      id: 'mirageLogging',
      label: 'Mirage logging',
    });
  }
```

Instead went with the macros from embroider for a more specific solution 
```javascript
import { dependencySatisfies } from '@embroider/macros';

if (dependencySatisfies('ember-qunit', '*')) {
  window.QUnit.config.urlConfig.push({ id: 'mirageLogging', label: 'Mirage logging', });
}
```

Used window because this is now in linted code. 


Good to know about importSync for future, but ended up not needing it here. 

Was also using ember-compatibility-helpers but embroider macros did the same thing, so changed that over as well.